adr/0024-starlight.md

```diff
@@ -0,0 +1,76 @@
+
+    - **Pros**:
+        - Fast performance.
+        - Easy to extend with custom components in any framework.
+        - Built-in support for MDX.
+        - Simplified build process.
+        - Better control over dependencies.
+        - Client-side static site search built-in.
+        - Excellent default theme.
+    - **Cons**:
+        - Newer project with a smaller ecosystem compared to others.
+        - Much fewer plugins and themes available compared to other static site generators.
+        - Only absolute URLs are supported for cross-site links (relative image URLs are supported).
+
+2. [**Hugo**](https://gohugo.io/): is another popular static site generator that is known for its speed and flexibility. It is written in Go and has a large ecosystem of themes and plugins.
+
+    - **Pros**:
+        - Fast performance.
+        - Large ecosystem of themes and plugins.
+        - By default no JavaScript is required.
+    - **Cons**:
+        - Fewer built-in features compared to others, much more of a DIY approach.
+        - Steeper learning curve compared to others.
+        - More complex build process.
+        - Not as easy to extend with custom components.
+        - Theme management is abysmal w/ a combination of Go modules, Git submodules, and NPM packages.
+        - Zero documentation themes the team liked, leading to the quandry of whether to build, and having to maintain, a custom theme.
+
+3. [**Material for MkDocs**](https://squidfunk.github.io/mkdocs-material/): is a popular static site generator geared towards project documentation. The Material for MkDocs theme provides a clean and responsive design.
+
+    - **Pros**:
+        - _Amazing_ default theme and developer experience for writing docs.
+        - Easy to use and extend.
+        - Large ecosystem of plugins.
+    - **Cons**:
+        - Not as flexible as Hugo or Starlight.
+        - Making custom components can be more challenging.
+        - Brings in Python dependencies (not a really huge detractor, but the team would like to reduce dependencies on different languages where possible).
+
+## Decision
+
+To simplify the development process and improve the performance of the documentation site, we have decided to migrate to Starlight. This is based upon a number of factors:
+
+- Small dependency footprint.
+- Fast performance.
+- Easy to extend with custom components.
+- Built-in support for MDX.
+- The majority of the features we need are built-in and require minimal configuration.
+  - Client-side static site search
+  - Sane spacing and typography
+  - Dark mode + light mode
+  - Syntax highlighting
+  - Responsive design + mobile-friendly
+  - Tabs + Admonitions
+
+## Consequences
```

Comment:
One of the primary design criteria, and the reason Hugo was ultimately chosen last year, was the ability to deploy the documentation in a secure environment through a hardened pipeline while avoiding the overhead of vulnerability management because there are no deployed dependencies. What is the plan for mitigating this with Starlight?

Reply:
Both tools result in a static website.

Not sure what you mean by `avoiding the overhead of vulnerability management`?

Secure pipelines can be built equally for JS or Go, using a pure JS pipeline actually reduces dependencies because you only follow 1 ecosystem (for the most part) - NPM.

Reply:
Because there are no deployed dependencies with Hugo, there is no need to spend effort resolving vulnerabilities as they appear.